src/certificate_transfer_integration.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helper class for trusting ca chains."""
+
+from typing import Callable, Union
+
+from charms.certificate_transfer_interface.v0.certificate_transfer import (
+    CertificateAvailableEvent,
+    CertificateRemovedEvent,
+    CertificateTransferRequires,
+)
+from ops import CharmBase, Object
+
+from constants import CA_CERT_DIR_PATH, CERTIFICATE_TRANSFER_NAME
+
+
+class CertTransfer(Object):
+    def __init__(
+        self,
+        charm: CharmBase,
+        container_name: str,
+        callback_fn: Callable,
+        cert_transfer_relation_name: str = CERTIFICATE_TRANSFER_NAME,
+        bundle_name: str = "ca-certificates.crt",
+    ):
+        super().__init__(charm, cert_transfer_relation_name)
+        self.charm = charm
+        self.container = charm.unit.get_container(container_name)
+        self.cert_transfer = CertificateTransferRequires(charm, cert_transfer_relation_name)
+        self.callback_fn = callback_fn
+        self.bundle_name = bundle_name
+
+        self.framework.observe(
+            self.cert_transfer.on.certificate_available, self._on_certificate_event
+        )
+        self.framework.observe(
+            self.cert_transfer.on.certificate_removed, self._on_certificate_event
+        )
+
+    @property
+    def ca_bundle(self) -> str:
```

Comment:
Just a question: what does this method do? Is it trying to get the CA cert from the "cert transfer" integration data?

Reply:
yes